docs(mastra-platform): document GitHub integration (#17984)
mastra-ai/mastra: docs/src/content/en/docs/mastra-platform/github.mdx

---
title: "GitHub integration | Mastra platform"
description: Connect a GitHub repository to a Mastra platform project to deploy Studio and Server on every push.
---

import Steps from "@site/src/components/Steps";
import StepItem from "@site/src/components/StepItem";

# GitHub integration on Mastra platform

> The GitHub integration links a Mastra platform project to a GitHub repository so that pushes to a branch deploy Studio and Server automatically. Use it when you want push-to-deploy instead of running `mastra studio deploy` or `mastra server deploy` from your terminal or CI.

Reviewer: ```suggestion
The GitHub integration links a Mastra platform project to a GitHub repository. When you push changes to your repository Studio and Server will automatically be deployed.
```

Author: applied in [f879e1d](https://github.com/mastra-ai/mastra/pull/17984/commits/f879e1d2b8f32c23b7872813d89d9ca1a4aadc54)

<Steps>
  <StepItem>
    In the [Mastra platform dashboard](https://projects.mastra.ai), open your organization settings. The page includes a **GitHub App** section.
  </StepItem>

  <StepItem>
    Select **Install GitHub App** and choose the GitHub account or organization to install on.

    If you don't own the GitHub organization, GitHub creates a pending **installation request** that an admin must approve. The dashboard tracks pending requests alongside completed installations.
  </StepItem>

  <StepItem>
    Select the repositories the App can access. You can scope to specific repositories or grant access to all repositories on the account.

    You can revoke or change repository access at any time from GitHub's App settings.
  </StepItem>
</Steps>

:::note
If GitHub later updates the App's required permissions, the dashboard surfaces an **outdated permissions** warning with an approval link. Repository operations return a `403` until an admin re-approves.
:::

### Pending installation requests

> When a non-admin requests the App on an organization, the request stays pending until an organization admin approves it on GitHub.com. The dashboard:

Reviewer: So does it work with self-hosted GitHub (enterprise) or only GitHub.com? If the latter this should be clarified earlier on

Author: Does not work with self-hosted GitHub, clarified in [f879e1d](https://github.com/mastra-ai/mastra/pull/17984/commits/f879e1d2b8f32c23b7872813d89d9ca1a4aadc54)
